From 4510d9f53bd77bc8b97d37c78b499d85d0a97d4f Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Sat, 3 Jan 2009 12:11:42 +0000
Subject: [PATCH 12/13] Add an implementation of umount(3) for ia64

Like x86_64 and parisc, ia64 lacks a single-argument umount syscall, so we
need to provide a stub implementation that just calls umount2(path, 0).
I don't know ia64 assembler, so this one is in C.
---
 ia64/Makefile.add     |    2 +-
 ia64/umount-wrapper.c |    5 +++++
 2 files changed, 6 insertions(+), 1 deletions(-)
 create mode 100644 ia64/umount-wrapper.c

Index: dietlibc-0.33~cvs20110918/ia64/Makefile.add
===================================================================
--- dietlibc-0.33~cvs20110918.orig/ia64/Makefile.add	2011-09-16 20:34:24.000000000 +0200
+++ dietlibc-0.33~cvs20110918/ia64/Makefile.add	2011-11-03 21:52:08.000000000 +0100
@@ -1,2 +1,2 @@
 VPATH:=ia64:syscalls.s:$(VPATH)
-LIBOBJ+=$(OBJDIR)/__time.o $(OBJDIR)/__waitpid.o $(OBJDIR)/__nice.o $(OBJDIR)/__alarm.o
+LIBOBJ+=$(OBJDIR)/__time.o $(OBJDIR)/__waitpid.o $(OBJDIR)/__nice.o $(OBJDIR)/__alarm.o $(OBJDIR)/umount-wrapper.o
Index: dietlibc-0.33~cvs20110918/ia64/umount-wrapper.c
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ dietlibc-0.33~cvs20110918/ia64/umount-wrapper.c	2011-11-03 21:52:08.000000000 +0100
@@ -0,0 +1,5 @@
+#include <sys/mount.h>
+
+int umount(const char *target) {
+  return umount2(target, 0);
+}
